plugin-ssh/che-plugin-ssh-ext-sshkey/src/test/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStoreTest.java

```diff
@@ -0,0 +1,232 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.ssh.server;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.eclipse.che.api.user.shared.dto.ProfileDescriptor;
+import org.hamcrest.core.StringContains;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Anton Korneta
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class UserProfileSshKeyStoreTest {
+
+    private static final String API_ENDPOINT = "http://0.0.0.0:2375";
+
+    private UserProfileSshKeyStore sshKeyStore;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void shouldGetThePrivateKeyFromUsersPreference() throws Exception {
+        sshKeyStore = spy(new UserProfileSshKeyStore(API_ENDPOINT));
+        final String privateKey = "private key";
+        final String publicKey = "public key";
+        doReturn(ImmutableMap.of("ssh.key.private." + privateKey, "ssh.key.public." + publicKey)).when(sshKeyStore).getSshKeys();
+        doNothing().when(sshKeyStore).updateSshKeys(any());
+
+        Set<String> result = sshKeyStore.getAll();
+
+        assertTrue(result.size() == 1 && result.contains(privateKey));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenGettingKeysFromUsersPreferences() throws Exception {
+        sshKeyStore = spy(new UserProfileSshKeyStore(API_ENDPOINT));
+        doThrow(IOException.class).when(sshKeyStore).getSshKeys();
+        thrown.expect(SshKeyStoreException.class);
+        thrown.expectMessage(new StringContains("Failed to get keys"));
+
+        sshKeyStore.getAll();
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenNoKeysInUsersPreferences() throws Exception {
+        sshKeyStore = spy(new UserProfileSshKeyStore(API_ENDPOINT));
+        doReturn(null).when(sshKeyStore).getSshKeys();
+        thrown.expect(SshKeyStoreException.class);
```

Comment:
I think you should separate _expectation_ section from _given_ and _action_ sections, this test case is not obvious